swap_meet/vendor.py

```diff
@@ -1,2 +1,125 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        """Adds an item to the instance's inventory and returns it."""
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        """
+        Removes items from the instance's inventory.
+
+        Parameter: 
+        item (str): Inventory item to be removed.
+
+        Returns:
+        bool: True if item is inventory, False otherwise.
+        """
+
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, specific_category):
+        """
+        Adds the items in the instance's inventory that are of a specific category.
+
+        Parameters: 
+        specific_category (str): A category.
+
+        Returns:
+        category_matches (list): Items in the inventory matching the category.
+        """
+
+        category_matches = []
+        for item in self.inventory:
+            if item.category == specific_category:
+                category_matches.append(item)
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
category_matches = [item for item in self.inventory if item.category == category]
```